websites/management/commands/markdown_cleaning/baseurl_rule_test.py

```diff
@@ -43,13 +43,13 @@ def get_markdown_cleaner(website_contents):
             R"This link should change [text title]({{< baseurl >}}/resources/path/to/file1) cool",
             R'This link should change {{< resource_link content-uuid-1 "text title" >}} cool',
         ),
-        (  # should not touch fragments
-            R"This link includes a fragment [text title]({{< baseurl >}}/resources/path/to/file1#some-fragment) cool",
+        (
             R"This link includes a fragment [text title]({{< baseurl >}}/resources/path/to/file1#some-fragment) cool",
+            R'This link includes a fragment {{< resource_link content-uuid-1 "text title" "some-fragment" >}} cool',
         ),
-        (  # should not touch fragments with / before #
-            R"This link includes a fragment with slash first [text title]({{< baseurl >}}/resources/path/to/file1/#some-fragment) cool",
+        (
             R"This link includes a fragment with slash first [text title]({{< baseurl >}}/resources/path/to/file1/#some-fragment) cool",
+            R'This link includes a fragment with slash first {{< resource_link content-uuid-1 "text title" "some-fragment" >}} cool',
```

Comment:
I was looking at the updated `resource_link.html` shortcode in `ocw-hugo-themes`:

```
{{- $uuid := index .Params 0 -}}
{{- $title := index .Params 1 -}}
{{- $anchor_id := index .Params 2 | default ""}}
{{- range where $.Site.Pages "Params.uid" $uuid -}}
    <a href="{{- .Permalink -}}{{- $anchor_id -}}">{{ $title }}</a>
{{- end -}}

```

We're fetching `fragment_id` there from the second argument, but it looks like the fragment itself here doesn't include the `#` character.  Are we expecting that character to be in the template, or passed in?  If we're not going to put it in the theme, this new argument could be used even to just put in any query string on the URL.  Either way, it doesn't currently look like the `#` will be preserved, or am I missing something?

Reply:
Ahhhhh good catch. I was testing this in studio but hadn't followed a link yet.

And yes, I agree—leaving the second arg flexible for `#` or `?` is good.